File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& n, int t) {
        unordered_map<int, int> mp;
        for(int i = 0; i < n.size(); i++) {
            if(mp.count(n[i])) {
                return {mp[n[i]], i + 1};
            }
            mp[t - n[i]] = i + 1;
        }
        return {};
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-1.rs
impl Solution {
pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
    let mut mp = HashMap::new();
    let target = target;
    for (i, n) in nums.iter().enumerate() {
        if let Some(k) = mp.get(n) {
            return vec![*k as i32, i as i32];
        };
        mp.insert(target - n, i);
    }
    vec![]
}
}
